### PR TITLE
fix: [ANDROAPP-3186] Show description icon when test is ellipsized

### PR DESCRIPTION
**Pull request by andresmr**
## Description
Please include a summary of the change and include the related jira issue if it exists.

[ jira issue ](https://jira.dhis2.org/browse/ANDROAPP-3186])

## Solution description
If this PR is a fix include a brief description on how the issue is solved.
## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [x] Smartphone Emulator
- [ ] Tablet Emulator
- [ ] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [ ] 8.X
- [x] 9.X - 10.X
- [ ] Other
## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

**Review by ferdyrod — APPROVED** [review]

**Review by ferdyrod — CHANGES_REQUESTED** [review]
testing this review state (omit this)

**Review by ferdyrod — COMMENTED** [review]
Testing review state (omit this)

**Review by ferdyrod — APPROVED** [review]

File: app/src/main/java/org/dhis2/data/forms/dataentry/fields/section/SectionHolder.kt [hunk 1]
```diff
@@ -69,51 +71,56 @@ class SectionHolder(
...
-            View.GONE
-        } else {
-            View.VISIBLE
+        viewModel.description().takeIf { it.isNullOrEmpty() }?.let {
+            formBinding.descriptionIcon.visibility = GONE
+            formBinding.sectionName.viewTreeObserver.addOnGlobalLayoutListener {
+                formBinding.sectionName.layout?.let {
+                    if (it.getEllipsisCount(0) > 0) {
+                        formBinding.descriptionIcon.visibility = VISIBLE
+                    }
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/data/forms/dataentry/fields/section/SectionHolder.kt:101]: Do you think it makes sense to extract this presentation logic? This is a very small example but when it gets bigger we cannot unit test :(

Comment (andresmr) [reply app/src/main/java/org/dhis2/data/forms/dataentry/fields/section/SectionHolder.kt:101]: yes, you are right, I will extract it. Thanks!.

**Review by JaimeToca — APPROVED** [review]

**Pull request merged**